core/src/SYCL/Kokkos_SYCL_Team.hpp

```diff
@@ -1016,8 +1016,17 @@ KOKKOS_INLINE_FUNCTION void single(
   const auto grange1          = item.get_local_range(1);
   const auto sg               = item.get_sub_group();
   if (item.get_local_id(1) == 0) lambda(val);
-  val = Kokkos::Impl::SYCLReduction::select_from_group(
-      sg, val, (sg.get_local_id() / grange1) * grange1);
+  // FIXME_SYCL oneAPI broke pointer support in sycl::select_from_group past the
+  // 2025.0.0 release. It's supposed to be fixed in the 2025.2.0 release.
+  if constexpr (std::is_pointer_v<ValueType>) {
+    uintptr_t tmp = reinterpret_cast<uintptr_t>(val);
+    tmp           = Kokkos::Impl::SYCLReduction::select_from_group(
+        sg, tmp, (sg.get_local_id() / grange1) * grange1);
+    val = reinterpret_cast<ValueType>(tmp);
+  } else {
+    val = Kokkos::Impl::SYCLReduction::select_from_group(
+        sg, val, (sg.get_local_id() / grange1) * grange1);
+  }
```

Comment:
Why not ifdef around this?

Reply:
Are you suggesting to only enable the workaround for compiler versions past 2025.0.0 (I haven't found/tried a version that has the issue fixed.) ? I don't feel strongly either way. This way has fewer separate code paths depending on compiler versions while we of course only want the old code in the end.

Reply:
We'll at least have a fallback implementation in place if it happens to break again in the future.

@masterleinad Do you plan on removing the `FIXME_SYCL` comment after 2025.2.0 becomes part of the minimum req?